x-pack/plugins/endpoint/common/types.ts

```diff
@@ -51,6 +51,29 @@ export class EndpointAppConstants {
   static MAX_LONG_INT = '9223372036854775807'; // 2^63-1
 }
 
+export interface NodeStats {
+  totalEvents: number;
+  totalAlerts: number;
+}
+
+export interface NodePagination {
+  nextChild?: string | null;
```

Comment:
❔ The `?` here marks it as "possibly undefined" right? How does the meaning differ when it's null? Is null for when it is "last page" and `undefined` is for when it hasn't been paged yet?

Reply:
ℹ️ Note to self to replace/merge some of the things in adjacencyMap with this so the meaning doesn't diverge

Reply:
> ❔ The `?` here marks it as "possibly undefined" right? How does the meaning differ when it's null? Is null for when it is "last page" and `undefined` is for when it hasn't been paged yet?

Yeah I think `null` indicates that there are no more events of that type to retrieve. So you've reached the last page. If there are more pages, it'll be the entity_id that should be in the next request. I think we're using `undefined` here so it just doesn't show up in the pagination section. I could be wrong though. I don't think it is denoting something specific.

Reply:
@jonathan-buttner @kqualters-elastic if there is special meaning to these interface fields, can we document it here?

Also, can you clarify what "you've reached the last page" means? Data could still be coming in right? So unless we know that there will never be any more data, we will probably need to assume that there is more.

Let me know if I'm wrong or unclear. I'm writing everything on a tablet and it's tedious 😅

Reply:
TODO ask about this

Reply:
@oatkiller Yeah that's a fair point that we won't ever know for sure that there isn't more data because we could have been received some since they last time we searched, but it's probably still useful to indicate that we know for sure that there is more data available right?

For example if we only retrieve a portion of the children for a node, by indicating that we know there are more children we could display an arrow or plus sign or something in the UI indicating to the user that there is definitely more children to retrieve.